cc_proxy.go

```diff
@@ -127,17 +130,44 @@ func (p *ccProxy) connectProxy(proxyURL string) (*client.Client, error) {
 	return client.NewClient(conn), nil
 }
 
+// start is the proxy start implementation for ccProxy.
+func (p *ccProxy) start(pod Pod) (int, string, error) {
+	if pod.config == nil {
+		return -1, "", fmt.Errorf("Pod config cannot be nil")
+	}
+
+	config, ok := newProxyConfig(*(pod.config)).(CCProxyConfig)
+	if !ok {
+		return -1, "", fmt.Errorf("Wrong proxy config type, should be CCProxyConfig type")
+	}
+
+	if config.Path == "" {
+		return -1, "", fmt.Errorf("Proxy path cannot be empty")
+	}
+
+	// construct the socket path the proxy instance will use
+	socketPath := filepath.Join(runStoragePath, pod.id, "proxy.sock")
+	uri := fmt.Sprintf("unix://%s", socketPath)
```

Comment:
I wonder if we have to ensure proxy's socket mode is 600

Reply:
This is just constructing a path to give to the proxy. The proxy itself creates the socket (with perms `0660`) here:

- https://github.com/clearcontainers/proxy/blob/master/proxy.go#L674

Reply:
ohh I see, thanks @jodh-intel for the link